File: CS/CustomPrintPreview/Grid/CustomGridViewPrinting.cs
using System;
using System.Drawing;
using DevExpress.Utils;
using System.Reflection;
using DevExpress.XtraTab;
using DevExpress.XtraGrid;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.Utils.Frames;
using DevExpress.Utils.Controls;
using DevExpress.XtraGrid.Design;
using DevExpress.XtraGrid.Columns;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Localization;
using DevExpress.XtraGrid.Views.Printing;
using DevExpress.Utils.Text;

namespace CustomPrintPreview.Grid {
    public partial class CustomGridViewPrinting :XtraFrame, IPrintDesigner {
        GridControl GridPreview;
        ImageCollection Images;
        int LockApply;
        string[] PrintFlags = new string[] {
            "PrintHeader", "PrintPreview", "PrintFooter", "PrintGroupFooter", "PrintHorzLines", 
            "PrintVertLines", "PrintFilterInfo", "PrintDetails", "AutoWidth", "ExpandAllDetails", 
            "ExpandAllGroups", "UsePrintStyles", "EnableAppearanceEvenRow", "EnableAppearanceOddRow", 
            "PrintBandHeader", "PrintSelectedRowsOnly"
        };
        GridColumn GroupColumn;

        CustomGridViewPrinting() : base(8) {
            InitializeComponent();
        }

        public CustomGridViewPrinting(GridControl grid, GridColumn groupColumn) : this() {
            PanelGridPreview.DockPadding.All = 4;
            GridPreview = grid;
            GridPreview.Dock = DockStyle.Fill;
            GridPreview.Parent = PanelGridPreview;
            Images = ImageHelper.CreateImageCollectionFromResources("DevExpress.XtraGrid.Printing.GridPrintOptions.png",
                typeof(GridViewDesigner).Assembly, new Size(16, 16));
            this.GroupColumn = groupColumn;
        }

        bool fAutoApply = true;
        public bool AutoApply {
            get { return fAutoApply; }
            set { fAutoApply = value; }
        }

        CustomGridView EditingView {
            get { return (CustomGridView)EditingObject; }
        }

        GridView CurrentView {
            get { return (GridView)GridPreview.MainView; }
        }

        public Size UserControlSize {
            get { return TextUtils.GetScaleSize(new Size(560, 345)); }
        }

        void OnTabPageOptionsPaint(object sender, PaintEventArgs e) {
            PaintTabPage(e.Graphics, 0, 8);
            PrintImage(12, e.Graphics);
            PrintImage(13, e.Graphics);
            PrintImage(15, e.Graphics);
        }

        void PrintImage(int index, Graphics graphics) {
            PrintImage(CheckEditByIndex(index), graphics, Images.Images[index]);
        }

        void PrintImage(CheckEdit editor, Graphics graphics, Image image) {
            if (editor == null) return;
            int top = editor.Top + (editor.Height - image.Size.Height) / 2;
            int x = 10;
            if (editor.Checked)
                graphics.DrawImage(image, new Rectangle(x, top, image.Width, image.Height));
            else ControlPaint.DrawImageDisabled(graphics, image, x, top, SystemColors.Control);
        }

        CheckEdit CheckEditByIndex(int index) {
            foreach (XtraTabPage tp in TabControl.TabPages)
                foreach (Control c in tp.Controls)
                    if (c is CheckEdit && Convert.ToInt32(c.Tag) == index)
                        return (CheckEdit)c;
            return null;
        }

        void OnCheckBoxCheckStateChanged(object sender, EventArgs e) {
            InvalidateImage((CheckEdit)sender);
        }

        void InvalidateImage(CheckEdit editor) {
            editor.Parent.Invalidate(new Rectangle(0, editor.Top, 50, editor.Height));
            ApplyOptions();
        }

        void ApplyOptions() {
            ApplyOptions(AutoApply);
        }

        protected virtual void ApplyOptions(bool setOptions) {
            if (LockApply != 0) return;
            for (int i = 0; i < PrintFlags.Length; i++) {
                CheckEdit chb = CheckEditByIndex(i);
                if (chb != null)
                    SetOptions.SetOptionValueByString(PrintFlags[i], setOptions ? EditingView.OptionsPrint :
                        CurrentView.OptionsPrint, chb.Checked);
            }
            EditingView.FireChangedInternal();
        }

        void OnCheckBoxHorzLinesCheckStateChanged(object sender, EventArgs e) {
            CurrentView.OptionsView.ShowHorzLines = CheckBoxHorzLines.Checked;
            InvalidateImage(CheckBoxHorzLines);
        }

        void OnCheckBoxVertLinesCheckStateChanged(object sender, EventArgs e) {
            CurrentView.OptionsView.ShowVertLines = CheckBoxVertLines.Checked;
            InvalidateImage(CheckBoxVertLines);
        }

        void OnCheckBoxFooterCheckStateChanged(object sender, EventArgs e) {
            CurrentView.OptionsView.ShowFooter = CheckBoxFooter.Checked;
            InvalidateImage(CheckBoxFooter);
        }

        void OnCheckBoxPreviewCheckStateChanged(object sender, EventArgs e) {
            CurrentView.OptionsView.ShowPreview = CheckBoxPreview.Checked;
            InvalidateImage(CheckBoxPreview);
        }

        void OnCheckBoxHeaderCheckStateChanged(object sender, EventArgs e) {
            CurrentView.OptionsView.ShowColumnHeaders = CheckBoxHeader.Checked;
            InvalidateImage(CheckBoxHeader);
        }

        void OnCheckBoxGroupFooterCheckStateChanged(object sender, EventArgs e) {
            if (GroupColumn != null) {
                if (CheckBoxGroupFooter.Checked)
                    GroupColumn.GroupIndex = 0;
                else {
                    GroupColumn.GroupIndex = -1;
                    GroupColumn.VisibleIndex = 3;
                }
            }
            CurrentView.ExpandAllGroups();
            CurrentView.FocusedRowHandle = -99;
            InvalidateImage(CheckBoxGroupFooter);
        }

        void OnCheckBoxStylesCheckStateChanged(object sender, EventArgs e) {
            OnCheckBoxCheckStateChanged(sender, e);
            InitViewStyles();
        }

        void OnCheckBoxDefaultStylesCheckStateChanged(object sender, EventArgs e) {
            OnCheckBoxStylesCheckStateChanged(sender, e);
            CheckBoxEvenRow.Enabled = CheckBoxOddRow.Enabled = CheckBoxDefaultStyles.Checked;
        }

        void InitViewStyles() {
            CurrentView.BeginUpdate();
            try {
                CurrentView.Appearance.Reset();
                if (CheckBoxDefaultStyles.Checked) {
                    GridViewPrintAppearances collection = ((BaseViewPrintInfo)typeof(
                        GridView).InvokeMember("CreatePrintInfoCore", BindingFlags.Instance | 
                        BindingFlags.NonPublic | BindingFlags.InvokeMethod, null, 
                        CurrentView, new object[] { 
                            new PrintInfoArgs(CurrentView) })).AppearancePrint as
                            GridViewPrintAppearances;
                    if (collection != null) {
                        CurrentView.Appearance.Assign(collection);
                        CurrentView.Appearance.HorzLine.Assign(collection.Lines);
                        CurrentView.Appearance.VertLine.Assign(collection.Lines);
                    }
                } else CurrentView.Appearance.Assign(EditingView.PaintAppearance);
                CurrentView.Appearance.FooterPanel.BorderColor = CurrentView.Appearance.FooterPanel.BackColor;
                CurrentView.Appearance.HeaderPanel.BorderColor = CurrentView.Appearance.HeaderPanel.BackColor;
                CurrentView.Appearance.GroupFooter.BorderColor = CurrentView.Appearance.GroupFooter.BackColor;
                CurrentView.OptionsView.EnableAppearanceEvenRow = CheckBoxDefaultStyles.Checked ?
                    CheckBoxEvenRow.Checked : EditingView.OptionsView.EnableAppearanceEvenRow;
                CurrentView.OptionsView.EnableAppearanceOddRow = CheckBoxDefaultStyles.Checked ?
                    CheckBoxOddRow.Checked : EditingView.OptionsView.EnableAppearanceOddRow;
                CurrentView.Appearance.Empty.BackColor = CurrentView.Appearance.Empty.BackColor2 =
                    Color.White;
            } finally { CurrentView.EndUpdate(); }
        }

        void PaintTabPage(Graphics graphics, int startIndex, int endIndex) {
            for (int i = startIndex; i < endIndex; i++)
                PrintImage(i, graphics);
        }

        void OnTabPageBehaviorPaint(object sender, PaintEventArgs e) {
            PaintTabPage(e.Graphics, 8, 11);
        }

        void OnCheckBoxAutoWidthCheckStateChanged(object sender, EventArgs e) {
            CurrentView.OptionsView.ColumnAutoWidth = CheckBoxAutoWidth.Checked;
            InvalidateImage(CheckBoxAutoWidth);
        }

        protected override string DescriptionText {
            get { return GridLocalizer.Active.GetLocalizedString(GridStringId.PrintDesignerDescription); }
        }

        public override void InitComponent() {
            LockApply--;
            try {
                for (int i = 0; i < PrintFlags.Length; i++) {
                    CheckEdit chb = CheckEditByIndex(i);
                    if (chb != null) {
                        chb.Checked = SetOptions.OptionValueByString(PrintFlags[i], EditingView.OptionsPrint);
                        if (chb.Checked)
                            chb.Checked = true;
                    }
                }
            } finally { LockApply++; }
            CurrentView.OptionsView.BeginUpdate();
            try {
                OnCheckBoxHeaderCheckStateChanged(CheckBoxHeader, EventArgs.Empty);
                OnCheckBoxPreviewCheckStateChanged(CheckBoxPreview, EventArgs.Empty);
                OnCheckBoxFooterCheckStateChanged(CheckBoxFooter, EventArgs.Empty);
                OnCheckBoxGroupFooterCheckStateChanged(CheckBoxGroupFooter, EventArgs.Empty);
                OnCheckBoxHorzLinesCheckStateChanged(CheckBoxHorzLines, EventArgs.Empty);
                OnCheckBoxVertLinesCheckStateChanged(CheckBoxVertLines, EventArgs.Empty);
                OnCheckBoxAutoWidthCheckStateChanged(CheckBoxAutoWidth, EventArgs.Empty);
                OnCheckBoxDefaultStylesCheckStateChanged(CheckBoxDefaultStyles, EventArgs.Empty);
                OnCheckBoxStylesCheckStateChanged(CheckBoxEvenRow, EventArgs.Empty);
                OnCheckBoxStylesCheckStateChanged(CheckBoxOddRow, EventArgs.Empty);
                OnCheckBoxCheckStateChanged(CheckBoxSelectedRows, EventArgs.Empty);
            } finally { CurrentView.OptionsView.EndUpdate(); }
            CurrentView.AppearancePrint.Assign(EditingView.AppearancePrint);
            CurrentView.OptionsPrint.Assign(EditingView.OptionsPrint);
            InitViewStyles();
            lbCaption.Text = GridLocalizer.Active.GetLocalizedString(GridStringId.PrintDesignerGridView);
            CurrentView.PaintStyleName = "MixedXP";
        }
        
#region IPrintDesigner Members

        void IPrintDesigner.ApplyOptions(bool setOptions) {
            ApplyOptions(setOptions);
        }

        void IPrintDesigner.HideCaption() {
            lbCaption.Visible = horzSplitter.Visible = false;
        }

        #endregion
    }
}

File: VB/CustomPrintPreview/Grid/CustomGridViewPrinting.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports DevExpress.Utils
Imports System.Reflection
Imports DevExpress.XtraTab
Imports DevExpress.XtraGrid
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.Utils.Frames
Imports DevExpress.Utils.Controls
Imports DevExpress.XtraGrid.Design
Imports DevExpress.XtraGrid.Columns
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Localization
Imports DevExpress.XtraGrid.Views.Printing
Imports DevExpress.Utils.Text

Namespace CustomPrintPreview.Grid
	Partial Public Class CustomGridViewPrinting
		Inherits XtraFrame
		Implements IPrintDesigner
		Private GridPreview As GridControl
		Private Images As ImageCollection
		Private LockApply As Integer
		Private PrintFlags() As String = { "PrintHeader", "PrintPreview", "PrintFooter", "PrintGroupFooter", "PrintHorzLines", "PrintVertLines", "PrintFilterInfo", "PrintDetails", "AutoWidth", "ExpandAllDetails", "ExpandAllGroups", "UsePrintStyles", "EnableAppearanceEvenRow", "EnableAppearanceOddRow", "PrintBandHeader", "PrintSelectedRowsOnly" }
		Private GroupColumn As GridColumn

		Private Sub New()
			MyBase.New(8)
			InitializeComponent()
		End Sub

		Public Sub New(ByVal grid As GridControl, ByVal groupColumn As GridColumn)
			Me.New()
			PanelGridPreview.DockPadding.All = 4
			GridPreview = grid
			GridPreview.Dock = DockStyle.Fill
			GridPreview.Parent = PanelGridPreview
			Images = ImageHelper.CreateImageCollectionFromResources("DevExpress.XtraGrid.Printing.GridPrintOptions.png", GetType(GridViewDesigner).Assembly, New Size(16, 16))
			Me.GroupColumn = groupColumn
		End Sub

		Private fAutoApply As Boolean = True
		Public Property AutoApply() As Boolean
			Get
				Return fAutoApply
			End Get
			Set(ByVal value As Boolean)
				fAutoApply = value
			End Set
		End Property

		Private ReadOnly Property EditingView() As CustomGridView
			Get
				Return CType(EditingObject, CustomGridView)
			End Get
		End Property

		Private ReadOnly Property CurrentView() As GridView
			Get
				Return CType(GridPreview.MainView, GridView)
			End Get
		End Property

		Public ReadOnly Property UserControlSize() As Size
			Get
				Return TextUtils.GetScaleSize(New Size(560, 345))
			End Get
		End Property

		Private Sub OnTabPageOptionsPaint(ByVal sender As Object, ByVal e As PaintEventArgs) Handles xtraTabPage1.Paint
			PaintTabPage(e.Graphics, 0, 8)
			PrintImage(12, e.Graphics)
			PrintImage(13, e.Graphics)
			PrintImage(15, e.Graphics)
		End Sub

		Private Sub PrintImage(ByVal index As Integer, ByVal graphics As Graphics)
			PrintImage(CheckEditByIndex(index), graphics, Images.Images(index))
		End Sub

		Private Sub PrintImage(ByVal editor As CheckEdit, ByVal graphics As Graphics, ByVal image As Image)
			If editor Is Nothing Then
				Return
			End If
			Dim top As Integer = editor.Top + (editor.Height - image.Size.Height) / 2
			Dim x As Integer = 10
			If editor.Checked Then
				graphics.DrawImage(image, New Rectangle(x, top, image.Width, image.Height))
			Else
				ControlPaint.DrawImageDisabled(graphics, image, x, top, SystemColors.Control)
			End If
		End Sub

		Private Function CheckEditByIndex(ByVal index As Integer) As CheckEdit
			For Each tp As XtraTabPage In TabControl.TabPages
				For Each c As Control In tp.Controls
					If TypeOf c Is CheckEdit AndAlso Convert.ToInt32(c.Tag) = index Then
						Return CType(c, CheckEdit)
					End If
				Next c
			Next tp
			Return Nothing
		End Function

		Private Sub OnCheckBoxCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkEdit2.CheckStateChanged, checkEdit1.CheckStateChanged, CheckBoxSelectedRows.CheckStateChanged, checkEdit4.CheckedChanged, checkEdit3.CheckStateChanged
			InvalidateImage(CType(sender, CheckEdit))
		End Sub

		Private Sub InvalidateImage(ByVal editor As CheckEdit)
			editor.Parent.Invalidate(New Rectangle(0, editor.Top, 50, editor.Height))
			ApplyOptions()
		End Sub

		Private Sub ApplyOptions()
			ApplyOptions(AutoApply)
		End Sub

        Protected Overridable Sub ApplyOptions(ByVal _setOptions As Boolean)
            If LockApply <> 0 Then
                Return
            End If
            For i As Integer = 0 To PrintFlags.Length - 1
                Dim chb As CheckEdit = CheckEditByIndex(i)
                If chb IsNot Nothing Then
                    If _setOptions Then
                        SetOptions.SetOptionValueByString(PrintFlags(i), EditingView.OptionsPrint, chb.Checked)
                    Else
                        SetOptions.SetOptionValueByString(PrintFlags(i), CurrentView.OptionsPrint, chb.Checked)
                    End If
                End If
            Next i
            EditingView.FireChangedInternal()
        End Sub

		Private Sub OnCheckBoxHorzLinesCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxHorzLines.CheckStateChanged
			CurrentView.OptionsView.ShowHorzLines = CheckBoxHorzLines.Checked
			InvalidateImage(CheckBoxHorzLines)
		End Sub

		Private Sub OnCheckBoxVertLinesCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxVertLines.CheckStateChanged
			CurrentView.OptionsView.ShowVertLines = CheckBoxVertLines.Checked
			InvalidateImage(CheckBoxVertLines)
		End Sub

		Private Sub OnCheckBoxFooterCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxFooter.CheckStateChanged
			CurrentView.OptionsView.ShowFooter = CheckBoxFooter.Checked
			InvalidateImage(CheckBoxFooter)
		End Sub

		Private Sub OnCheckBoxPreviewCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxPreview.CheckStateChanged
			CurrentView.OptionsView.ShowPreview = CheckBoxPreview.Checked
			InvalidateImage(CheckBoxPreview)
		End Sub

		Private Sub OnCheckBoxHeaderCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxHeader.CheckStateChanged
			CurrentView.OptionsView.ShowColumnHeaders = CheckBoxHeader.Checked
			InvalidateImage(CheckBoxHeader)
		End Sub

		Private Sub OnCheckBoxGroupFooterCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxGroupFooter.CheckStateChanged
			If GroupColumn IsNot Nothing Then
				If CheckBoxGroupFooter.Checked Then
					GroupColumn.GroupIndex = 0
				Else
					GroupColumn.GroupIndex = -1
					GroupColumn.VisibleIndex = 3
				End If
			End If
			CurrentView.ExpandAllGroups()
			CurrentView.FocusedRowHandle = -99
			InvalidateImage(CheckBoxGroupFooter)
		End Sub

		Private Sub OnCheckBoxStylesCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxOddRow.CheckStateChanged, CheckBoxEvenRow.CheckStateChanged
			OnCheckBoxCheckStateChanged(sender, e)
			InitViewStyles()
		End Sub

		Private Sub OnCheckBoxDefaultStylesCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxDefaultStyles.CheckStateChanged
			OnCheckBoxStylesCheckStateChanged(sender, e)
			CheckBoxOddRow.Enabled = CheckBoxDefaultStyles.Checked
			CheckBoxEvenRow.Enabled = CheckBoxOddRow.Enabled
		End Sub

		Private Sub InitViewStyles()
			CurrentView.BeginUpdate()
			Try
				CurrentView.Appearance.Reset()
				If CheckBoxDefaultStyles.Checked Then
					Dim collection As GridViewPrintAppearances = TryCast((CType(GetType(GridView).InvokeMember("CreatePrintInfoCore", BindingFlags.Instance Or BindingFlags.NonPublic Or BindingFlags.InvokeMethod, Nothing, CurrentView, New Object() { New PrintInfoArgs(CurrentView) }), BaseViewPrintInfo)).AppearancePrint, GridViewPrintAppearances)
					If collection IsNot Nothing Then
						CurrentView.Appearance.Assign(collection)
						CurrentView.Appearance.HorzLine.Assign(collection.Lines)
						CurrentView.Appearance.VertLine.Assign(collection.Lines)
					End If
				Else
					CurrentView.Appearance.Assign(EditingView.PaintAppearance)
				End If
				CurrentView.Appearance.FooterPanel.BorderColor = CurrentView.Appearance.FooterPanel.BackColor
				CurrentView.Appearance.HeaderPanel.BorderColor = CurrentView.Appearance.HeaderPanel.BackColor
				CurrentView.Appearance.GroupFooter.BorderColor = CurrentView.Appearance.GroupFooter.BackColor
				If CheckBoxDefaultStyles.Checked Then
					CurrentView.OptionsView.EnableAppearanceEvenRow = CheckBoxEvenRow.Checked
				Else
					CurrentView.OptionsView.EnableAppearanceEvenRow = EditingView.OptionsView.EnableAppearanceEvenRow
				End If
				If CheckBoxDefaultStyles.Checked Then
					CurrentView.OptionsView.EnableAppearanceOddRow = CheckBoxOddRow.Checked
				Else
					CurrentView.OptionsView.EnableAppearanceOddRow = EditingView.OptionsView.EnableAppearanceOddRow
				End If
				CurrentView.Appearance.Empty.BackColor2 = Color.White
				CurrentView.Appearance.Empty.BackColor = CurrentView.Appearance.Empty.BackColor2
			Finally
				CurrentView.EndUpdate()
			End Try
		End Sub

		Private Sub PaintTabPage(ByVal graphics As Graphics, ByVal startIndex As Integer, ByVal endIndex As Integer)
			For i As Integer = startIndex To endIndex - 1
				PrintImage(i, graphics)
			Next i
		End Sub

		Private Sub OnTabPageBehaviorPaint(ByVal sender As Object, ByVal e As PaintEventArgs) Handles xtraTabPage2.Paint
			PaintTabPage(e.Graphics, 8, 11)
		End Sub

		Private Sub OnCheckBoxAutoWidthCheckStateChanged(ByVal sender As Object, ByVal e As EventArgs) Handles CheckBoxAutoWidth.CheckStateChanged
			CurrentView.OptionsView.ColumnAutoWidth = CheckBoxAutoWidth.Checked
			InvalidateImage(CheckBoxAutoWidth)
		End Sub

		Protected Overrides ReadOnly Property DescriptionText() As String
			Get
				Return GridLocalizer.Active.GetLocalizedString(GridStringId.PrintDesignerDescription)
			End Get
		End Property

		Public Overrides Sub InitComponent()
			LockApply -= 1
			Try
				For i As Integer = 0 To PrintFlags.Length - 1
					Dim chb As CheckEdit = CheckEditByIndex(i)
					If chb IsNot Nothing Then
						chb.Checked = SetOptions.OptionValueByString(PrintFlags(i), EditingView.OptionsPrint)
						If chb.Checked Then
							chb.Checked = True
						End If
					End If
				Next i
			Finally
				LockApply += 1
			End Try
			CurrentView.OptionsView.BeginUpdate()
			Try
				OnCheckBoxHeaderCheckStateChanged(CheckBoxHeader, EventArgs.Empty)
				OnCheckBoxPreviewCheckStateChanged(CheckBoxPreview, EventArgs.Empty)
				OnCheckBoxFooterCheckStateChanged(CheckBoxFooter, EventArgs.Empty)
				OnCheckBoxGroupFooterCheckStateChanged(CheckBoxGroupFooter, EventArgs.Empty)
				OnCheckBoxHorzLinesCheckStateChanged(CheckBoxHorzLines, EventArgs.Empty)
				OnCheckBoxVertLinesCheckStateChanged(CheckBoxVertLines, EventArgs.Empty)
				OnCheckBoxAutoWidthCheckStateChanged(CheckBoxAutoWidth, EventArgs.Empty)
				OnCheckBoxDefaultStylesCheckStateChanged(CheckBoxDefaultStyles, EventArgs.Empty)
				OnCheckBoxStylesCheckStateChanged(CheckBoxEvenRow, EventArgs.Empty)
				OnCheckBoxStylesCheckStateChanged(CheckBoxOddRow, EventArgs.Empty)
				OnCheckBoxCheckStateChanged(CheckBoxSelectedRows, EventArgs.Empty)
			Finally
				CurrentView.OptionsView.EndUpdate()
			End Try
			CurrentView.AppearancePrint.Assign(EditingView.AppearancePrint)
			CurrentView.OptionsPrint.Assign(EditingView.OptionsPrint)
			InitViewStyles()
			lbCaption.Text = GridLocalizer.Active.GetLocalizedString(GridStringId.PrintDesignerGridView)
			CurrentView.PaintStyleName = "MixedXP"
		End Sub

#Region "IPrintDesigner Members"

		Private Sub IPrintDesigner_ApplyOptions(ByVal setOptions As Boolean) Implements IPrintDesigner.ApplyOptions
			ApplyOptions(setOptions)
		End Sub

		Private Sub HideCaption() Implements IPrintDesigner.HideCaption
			horzSplitter.Visible = False
			lbCaption.Visible = horzSplitter.Visible
		End Sub

		#End Region
	End Class
End Namespace
